sdcm/utils/decorators.py

```diff
@@ -373,6 +375,24 @@ def wrapper(*args, **kwargs):
     return wrapper
 
 
+def critical_on_capacity_issues(func: callable) -> callable:
+    """
+    Decorator to end the test with a critical event due to capacity issues
+    This should be used when a failure would leave the cluster in an inconsistent topology state
+    """
+    @wraps(func)
+    def wrapper(*args, **kwargs):
+        try:
+            return func(*args, **kwargs)
+        except ClientError as ex:
+            if "InsufficientInstanceCapacity" in str(ex):
+                TestFrameworkEvent(source=callable.__name__,
+                                   message=f"Test failed due to capacity issues: {ex}",
+                                   severity=Severity.CRITICAL).publish()
+            raise
```

Comment:
will this mean that it will raise even if critical event is published? If so, wont it generate two errors?

Reply:
it would, one would be nemesis failure, and on this critical event., and on the failure of where the test stopped.

critical event are in most cases not coming alone.

we could minimize it by raising a specific exception, that the nemesis loop can identify and change the nemesis event from error to critical

but I wouldn't go into that extent, not until we have more similar use case that dims it.